crates/tako/src/internal/worker/resources/allocator.rs

```diff
@@ -5,23 +5,23 @@ use crate::internal::common::resources::{
     ResourceAllocation, ResourceAllocations, ResourceId, ResourceVec,
 };
 use crate::internal::server::workerload::WorkerResources;
-use crate::internal::worker::resources::counts::{
-    resource_count_add_at, ResourceCount, ResourceCountVec,
-};
+use crate::internal::worker::resources::concise::{ConciseFreeResources, ConciseResourceState};
 use crate::internal::worker::resources::map::ResourceLabelMap;
 use crate::internal::worker::resources::pool::ResourcePool;
-use crate::resources::{Allocation, ResourceAmount, ResourceDescriptor, ResourceMap};
-use smallvec::smallvec;
+use crate::resources::{
+    Allocation, ResourceAmount, ResourceDescriptor, ResourceMap, ResourceUnits,
+};
+use std::rc::Rc;
 use std::time::Duration;
 
 pub struct ResourceAllocator {
     pools: ResourceVec<ResourcePool>,
-    free_resources: ResourceCountVec,
+    free_resources: ConciseFreeResources,
     remaining_time: Option<Duration>,
 
     blocked_requests: Vec<BlockedRequest>,
     higher_priority_blocked_requests: usize,
-    running_tasks: Vec<ResourceCountVec>, // TODO: Rework on multiset?
+    running_tasks: Vec<Rc<Allocation>>, // TODO: Rework on multiset?
```

Comment:
Originaly there was two types of allocation "just description" (e.g.: 5 cpus) and full allocation (indices: [0,10,5,1,4]). Allocator returned full allocation and remembers only the light one. With fraction resources, allocator is no longer two-phased (first do a light allocation and after that full) [technicly it is possible, but the description 
of allocation cannot be just a number with fractional resources and it almost look like full alocation, so description allocations were removed].
So allocation is now single phased. Allocator now returns an allocation but also remembers the full alocation. To avoid coping, it is wrapped to to Rc. (It ale makes faster the dealocation since to find the right allocation, we need to just check pointers and not full allocations.